tsl/test/shared/sql/constify_now.sql.in

```diff
@@ -189,3 +189,47 @@ SELECT * FROM
   LATERAL (SELECT m2.time FROM devices LIMIT 1) as subq_1
 WHERE subq_1.time > m1.time;
 
+CREATE TABLE logged_data (
+    rawtag_id   INTEGER,
+    timestamp   TIMESTAMP WITH TIME ZONE NOT NULL,
+    value       REAL,
+    site_id     INTEGER
+);
+
+SELECT table_name FROM create_hypertable('logged_data', 'timestamp');
+
+INSERT INTO logged_data (rawtag_id, timestamp, value, site_id)
+VALUES
+(1, '2023-01-01'::timestamptz, 13, 1),
+(1, '2023-01-07'::timestamptz, 13, 1),
+(1, '2023-01-10'::timestamptz, 13, 1),
+(1, '2023-01-15'::timestamptz, 13, 1),
+(1, '2023-01-20'::timestamptz, 15, 1),
+(2, '2023-01-01'::timestamptz, 90, 1),
+(3, '2023-01-07'::timestamptz, 2, 1),
+(4, '2023-01-10'::timestamptz, 13, 3),
+(2, '2023-01-15'::timestamptz, 13, 4),
+(5, '2023-01-20'::timestamptz, 13, 1);
+
+-- four chunks, all of them should be excluded at plantime
+SELECT COUNT(*) FROM show_chunks('logged_data');
+
+SET timescaledb.current_timestamp_mock TO '2024-01-01 0:30:00+00';
+
+SET timescaledb.enable_chunk_append TO true;
+SET timescaledb.enable_constraint_aware_append TO true;
+
+-- no startup exclusion should be happening for any of the queries below
+EXPLAIN (ANALYZE, SUMMARY OFF, TIMING OFF) SELECT * FROM logged_data WHERE
+timestamp BETWEEN now() AND now() - interval '1 day'
+AND rawtag_id = 1 ORDER BY "timestamp" ASC;
```

Comment:
So what are you getting here, something like
```
(BoolExprAnd
    (OpExpr (= rawtag_id 1))
    (BoolExprAnd
        (>= time now()) (<= time ...)))
```
Top-level AND should normally be flattened into a list, but  `ts_constify_now` is called before it happens, so this is why we have this problem?

Reply:
I haven’t yet found where the flattening happens, but yes that is how the expression looks: (pprint(node) in ts_constify_now)
```
{BOOLEXPR 
   :boolop and 
   :args (
      {BOOLEXPR 
      :boolop and 
      :args (
         {OPEXPR 
         :opno 1325 
         :opfuncid 1156 
         :opresulttype 16 
         :opretset false 
         :opcollid 0 
         :inputcollid 0 
         :args (
            {VAR 
            :varno 1 
            :varattno 1 
            :vartype 1184 
            :vartypmod -1 
            :varcollid 0 
            :varlevelsup 0 
            :varnosyn 1 
            :varattnosyn 1 
            :location 26
            }
            {OPEXPR 
            :opno 1329 
            :opfuncid 1190 
            :opresulttype 1184 
            :opretset false 
            :opcollid 0 
            :inputcollid 0 
            :args (
               {FUNCEXPR 
               :funcid 1299 
               :funcresulttype 1184 
               :funcretset false 
               :funcvariadic false 
               :funcformat 0 
               :funccollid 0 
               :inputcollid 0 
               :args <> 
               :location 39
               }
               {CONST 
               :consttype 1186 
               :consttypmod -1 
               :constcollid 0 
               :constlen 16 
               :constbyval false 
               :constisnull false 
               :location 56 
               :constvalue 16 [ 0 14 39 7 0 0 0 0 0 0 0 0 0 0 0 0 ]
               }
            )
            :location 45
            }
         )
         :location 31
         }
         {OPEXPR 
         :opno 1323 
         :opfuncid 1155 
         :opresulttype 16 
         :opretset false 
         :opcollid 0 
         :inputcollid 0 
         :args (
            {VAR 
            :varno 1 
            :varattno 1 
            :vartype 1184 
            :vartypmod -1 
            :varcollid 0 
            :varlevelsup 0 
            :varnosyn 1 
            :varattnosyn 1 
            :location 26
            }
            {FUNCEXPR 
            :funcid 1299 
            :funcresulttype 1184 
            :funcretset false 
            :funcvariadic false 
            :funcformat 0 
            :funccollid 0 
            :inputcollid 0 
            :args <> 
            :location 68
            }
         )
         :location 31
         }
      )
      :location 31
      }
      {OPEXPR 
      :opno 96 
      :opfuncid 65 
      :opresulttype 16 
      :opretset false 
      :opcollid 0 
      :inputcollid 0 
      :args (
         {VAR 
         :varno 1 
         :varattno 2 
         :vartype 23 
         :vartypmod -1 
         :varcollid 0 
         :varlevelsup 0 
         :varnosyn 1 
         :varattnosyn 2 
         :location 78
         }
         {CONST 
         :consttype 23 
         :consttypmod -1 
         :constcollid 0 
         :constlen 4 
         :constbyval true 
         :constisnull false 
         :location 82 
         :constvalue 4 [ 1 0 0 0 0 0 0 0 ]
         }
      )
      :location 80
      }
   )
   :location 74
   }
```

Reply:
I believe the flattening happens in `eval_const_expressions`, which gets called after ts_constify_now.